### PR TITLE
616-logging-dont-log-actions-if-if-prevents-them-from-executing

### PR DESCRIPTION
**Pull request by mborodii-prog**
### Enhanced Wrangle Execution Logging
**Summary**
This PR introduces comprehensive improvements to wrangle execution logging, providing users with clear visibility into data transformations and execution flow.

***Key Features***
**Actual Column Names in Logs**
Replaces generic 'Dynamic' placeholder with actual column names created by each wrangle
Supports wildcard expansion (Col* → Col1, Col2, Col3)
Handles regex patterns ("regex:Out([1-2])": "Renamed\\1" → Out1, Out2)
Processes mixed output formats (strings, lists, dictionaries)
**Skip Condition Logging**
Logs when wrangles are skipped due to if conditions
Clear format: : Skipped Wrangle :: wrangle_name :: if condition was false
Provides visibility into conditional execution flow
**Enhanced Log Format**
Improved readability with comma-separated column lists
Converts list inputs to readable strings (['col1'] → col1)
Maintains special handling for log wrangle to preserve existing behavior
***Implementation***
Core Changes
Modified _execute_wrangles() in wrangles/recipe.py recipe.py:525 recipe.py:818-829
Added skip logging before continue statements in if condition checks recipe.py:570-586
Enhanced output processing with wildcard and regex pattern expansion
Test Coverage
Comprehensive test suite in tests/recipes/wrangles/test_main.py
Covers all logging scenarios: basic execution, dynamic output, regex patterns, skip conditions, mixed types
Ensures robust handling of edge cases and error conditions

**Benefits**
Debugging: Clear visibility into what columns each wrangle actually creates
Monitoring: Better understanding of execution flow and skipped operations
Development: Easier to track data transformations through complex recipes
Documentation: Self-documenting execution logs for audit trails
Example
Before:

: Wrangling :: split.text :: Col1 >> Dynamic  
: Wrangling :: split.dictionary :: ['col1'] >> [{'regex:Out([1-2])': 'Renamed\\1'}]  
After:

: Wrangling :: split.text :: Col1 >> Col1, Col2, Col3  
: Wrangling :: split.dictionary :: col1 >> Out1, Out2  
: Skipped Wrangle :: convert.case :: if condition was false  
**Notes**
The log wrangle maintains its original : Dataframe :: format to preserve existing functionality and test compatibility .

### CODE REVIEW

**Comment by thomasstvr** [comment]
@mborodii-prog just before the continue, on the if evaluation, you could add something like this:

```python
_logging.info(f": {wrangle} skipped due to not passing the if statement.")
```

Also, it would be nice to let user's know exactly which one was skipped, like mentioned in my comment on issue #603. Let's keep that part separate from this though, so you don't have to figure out the location of the wrangle for now.

File: wrangles/recipe.py [hunk 1]
```diff
@@ -821,6 +822,49 @@ def _execute_wrangles(
                     # Run a second pass of df.fillna() in order to fill NaT's (not picked up before) with zeros
                     # Could also use _pandas.api.types.is_datetime64_any_dtype(df) as a check
                     df = df.fillna('0')
+                    if wrangle != 'log':  
+                        # Determine what columns were actually produced for logging
+                        if 'output' in params:  
+                            if isinstance(params['output'], list):
```
Comment (thomasstvr) [inline wrangles/recipe.py]: Wildcards are often passed to the output as a list, allow wildcard expansion for these cases. Limit to lists of length 1.

Comment (mborodii-prog) [reply wrangles/recipe.py]: Fixed.
